game/templates/game/leaderboard.html

```
@@ -2,194 +2,203 @@
+        <!-- Real-time client-side search box -->
+        <div class="search-container">
+            <span class="search-icon">
+                <svg width="18" height="18" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line></svg>
+            </span>
+            <input type="text" id="leaderboard-search" class="search-input" placeholder="Search by username..." aria-label="Search by username">
+        </div>
+    </div>
+
+    <!-- Tabs Header -->
+    <div class="tabs-container">
+        <button class="tab-button active" data-tab="puzzle">🏆 Puzzle Leaderboard</button>
```

HTMLHint (1.9.2)

[warning] The type attribute must be present on <button> elements.

(button-type-require)